exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/ReturningTests.kt

```diff
@@ -157,4 +160,27 @@ class ReturningTests : DatabaseTestsBase() {
             assertTrue { result3.all { it == 0.0 } }
         }
     }
+
+    object User : IdTable<String>("user") {
+        val username = varchar("username", 128)
+        override val primaryKey: PrimaryKey = PrimaryKey(username)
+        override val id: Column<EntityID<String>> = username.entityId()
+    }
+
+    @Test
+    fun testBatchInsertReturningCustomPrimaryKey() {
+        withTables(TestDB.enabledDialects() - returningSupportedDb, User) {
+            val result1 = User.batchInsert(listOf("batman")) { username ->
+                this[User.username] = username
+            }.single()
+            assertEquals("batman", result1[User.id].value)
+            assertEquals("batman", result1[User.username])
+
+            val result2 = User.batchInsert(listOf("superman")) { username ->
+                this[User.username] = username
+            }.single()
+            assertEquals("superman", result2[User.username])
+            assertEquals("superman", result2[User.id].value)
```

Comment:
Fails, because `result2[User.username]` caches the return value as a `String`, meaning `result2[User.id].value` produces a `ClassCastException`

Reply:
Arguably should be a separate PR, because this is not related to SQLite specifically - this fails on any database

Reply:
Created another https://youtrack.jetbrains.com/issue/EXPOSED-409/Custom-primary-key.-Access-to-the-primary-key-fails-with-ClassCastException issue for the second problem. It's clear why it happens, I'll try to fix it a bit later.